site/src/components/Filter/filter.tsx

```diff
@@ -33,45 +34,75 @@ export type PresetFilter = {
 
 type FilterValues = Record<string, string | undefined>;
 
+type UseFilterConfig = {
+  /**
+   * The fallback value to use in the event that no filter params can be parsed
+   * from the search params object. This value is allowed to change on
+   * re-renders.
+   */
+  fallbackFilter?: string;
+  searchParamsResult: ReturnType<typeof useSearchParams>;
+  onUpdate?: (newValue: string) => void;
+};
+
+const useFilterParamsKey = "filter";
+
 export const useFilter = ({
-  initialValue = "",
-  onUpdate,
+  fallbackFilter = "",
   searchParamsResult,
-}: {
-  initialValue?: string;
-  searchParamsResult: ReturnType<typeof useSearchParams>;
-  onUpdate?: () => void;
-}) => {
+  onUpdate,
+}: UseFilterConfig) => {
   const [searchParams, setSearchParams] = searchParamsResult;
-  const query = searchParams.get("filter") ?? initialValue;
-  const values = parseFilterQuery(query);
-
-  const update = (values: string | FilterValues) => {
-    if (typeof values === "string") {
-      searchParams.set("filter", values);
-    } else {
-      searchParams.set("filter", stringifyFilter(values));
-    }
+  const query = searchParams.get(useFilterParamsKey) ?? fallbackFilter;
+
+  // Using ref approach to be defensive against useSearchParams not returning a
+  // stable set function on each render. Effect must run before query effect
+  const setSearchParamsRef = useRef(setSearchParams);
```

Comment:
Interesting, I thought `setSearchParams` would be stable between render 🤔

Reply:
could `useEffectEvent` be used here?

Reply:
I would've thought so, too, but I had some issues with it making my `useEffect` calls run too often, and I also saw other people complaining about it on StackOverflow